src/Pages/Shared/ConfigMapSecret/validations.ts

```diff
@@ -0,0 +1,297 @@
+import YAML from 'yaml'
+
+import {
+    CMSecretExternalType,
+    UseFormValidation,
+    UseFormValidations,
+    YAMLStringify,
+} from '@devtron-labs/devtron-fe-common-lib'
+
+import { PATTERNS } from '@Config/constants'
+import { ValidationRules } from '@Components/cdPipeline/validationRules'
+
+import { CONFIG_MAP_SECRET_NO_DATA_ERROR, CONFIG_MAP_SECRET_YAML_PARSE_ERROR, SECRET_TOAST_INFO } from './constants'
+import { getESOSecretDataFromYAML, hasESO } from './utils'
+import { CMSecretYamlData, ConfigMapSecretUseFormProps } from './types'
+
+/**
```

Comment:
NOTE: (Can pick in future) One validation i found while testing, was for key name we should allow scoped variables to be added for cm/cs, since deployment template allows that.
cc @abhibhaw

Reply:
You mean adding scoped variables as key of a CM?

Reply:
Yes

Reply:
Yes we can see this in future